src/lib/hidden_node_test.ts

```diff
@@ -0,0 +1,47 @@
+import {hiddenNode} from './hidden_node';
+import {html, render} from 'lit-html';
+
+describe('The hiddenNode function for rule 2A', () => {
+
+  let container: HTMLElement;
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  it('returns empty string for hidden elems that aren\'t referenced', () => {
+    render(html`<div id="foo" hidden>Hello world</div>`, container);
+    const elem = document.getElementById('foo');
+    if (elem) {
```

Comment:
instead of checking if the `elem` is null, you can also just assert it since you know that it cannot be null:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
Really useful, thanks!